File: views/stats/map.js
function(doc) {
  emit('total items', 1);
  emit('total count', doc.count);

  if (doc.state == 'white') {
    emit('whitelisted items', 1);
    emit('whitelisted count', doc.count);
  } else if (doc.state == 'grey') {
    emit('greylisted items', 1);
    emit('greylisted count', doc.count);
  }
}

File: views/stats/map.erl
fun({Doc}) ->
    S = proplists:get_value(<<"state">>, Doc),
    %% C = proplists:get_value(<<"count">>, Doc),

    C = case proplists:get_value(<<"count">>, Doc) of
      C1 when is_integer(C1) -> C1;
      C1 when is_list(C1) -> list_to_integer(C1);
      C1 when is_binary(C1) -> list_to_integer(binary_to_list(C1));
      C1 -> 0
    end,

    Emit(<<"items total">>, 1),
    Emit(<<"count total">>, C),

    case S of
      <<"white">> ->
        Emit(<<"items white">>, 1),
        Emit(<<"count white">>, C);
      <<"grey">> ->
        Emit(<<"items grey">>, 1),
        Emit(<<"count grey">>, C)
    end
end.
